README.md

```diff
@@ -20,11 +20,102 @@ This repository contains the core Mobius framework and add-ons for common develo
 | 🛠 Xcode    | 10.1+       |
 | 🐦 Language | Swift 4.2  |
 
-## Usage
+## Usage in Your Project
 Pull in Mobius.swift as a dependency. Either as a submodule or using [Carthage](https://github.com/Carthage/Carthage).
 
 Build the project and link with the frameworks.
 
+## Mobius in Action - Building a Counter
+
+The goal of Mobius is to give you better control over your application state. You can think of your state as a snapshot of all the current values of the variables in your application. In Mobius, we take encapsulate all of the state in a data-structure which we call the *Model*.
+
+The *Model* can be represented by whatever type you like. Since we're building a counter, we'll be able to encapsulate all of our state in an `Int`:
+
+```swift
+typealias CounterModel = Int
+```
+
+Mobius does not let you manipulate the state directly. In order to change the state, you have to send the framework messages saying what you want to do. We call these messages *Events*. In our case, we'll want to be able to increment and decrement our counter. Let's use an `enum` to define these cases:
+```swift
+enum CounterEvent {
+    case increment
+    case decrement
+}
+```
+
+Now that we have a *Model* and some *Event*s, we'll need to give Mobius a set of rules which it can use to update the state on our behalf. We do this by giving the framework a function which it can call with its latest *Model* and any incomming *Event*, in order to generate a new *Model*:
```

Comment:
"will be sequentially called with every incoming *Event* and the most recent *Model*, in order to generate the next *Model*"